src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -314,17 +318,60 @@ void WriteTypeMappings (List<TypeDefinition> types)
 			using (var gen = UseSharedRuntime
 				? new TypeNameMapGenerator (types, Log.LogDebugMessage)
 			        : new TypeNameMapGenerator (ResolvedAssemblies.Select (p => p.ItemSpec), Log.LogDebugMessage)) {
-				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.jm"), gen.WriteJavaToManaged);
-				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.mj"), gen.WriteManagedToJava);
+				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.jm"), "jm", gen.WriteJavaToManaged);
+				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.mj"), "mj", gen.WriteManagedToJava);
 			}
 		}
 
-		void UpdateWhenChanged (string path, Action<Stream> generator)
+		void UpdateWhenChanged (string path, string type, Action<Stream> generator)
 		{
 			using (var stream = new MemoryStream ()) {
 				generator (stream);
 				MonoAndroidHelper.CopyIfStreamChanged (stream, path);
 			}
+
+			string dataFilePath = $"{path}.inc";
+			using (var stream = new NativeAssemblyDataStream ()) {
+				generator (stream);
+				stream.Flush ();
+				MonoAndroidHelper.CopyIfStreamChanged (stream, dataFilePath);
+
+				var generatedFiles = new List <ITaskItem> ();
+				string mappingFieldName = $"{type}_typemap";
+				string dataFileName = Path.GetFileName (dataFilePath);
+				NativeAssemblerTargetProvider asmTargetProvider;
+				foreach (string abi in SupportedAbis.Split (new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries)) {
+					switch (abi.Trim ()) {
+						case "armeabi-v7a":
+							asmTargetProvider = new ARMNativeAssemblerTargetProvider (false);
```

Comment:
An attempted coding convention is the use of named parameters for boolean values, because otherwise I have no idea what `false` means without going to the `ARMNativeAssemblerTargetProvider` constructor definition.

I find this much easier to understand:

```suggestion
							asmTargetProvider = new ARMNativeAssemblerTargetProvider (is64Bit: false);
```